### PR TITLE
Add support for MIME type application/x-directory

### PR DESCRIPTION
**Pull request by marklnichols**

### CODE REVIEW

File: src/client/import.ml [hunk 1]
```diff
@@ -440,23 +440,27 @@ end
...
 module Blob = struct
   module Format = struct
-    type t = [`Image | `Video] * string
+    type t = [`Image | `Video | `Appx] * string
 
     let gif = (`Image, "gif")
     let jpeg = (`Image, "jpeg")
     let png = (`Image, "png")
     let mp4 = (`Video, "mp4")
+    let appx = (`Appx, "appx")
```
Comment (smondet) [inline src/client/import.ml:449]: shouldn't this be `` (`Appx, "x-directory") ``?

File: src/client/token_viewer.ml [hunk 2]
```diff
@@ -275,7 +275,8 @@ let show_multimedia :
                     ( match mm.format with
                     | `Image, "svg+xml" -> "Vector Graphics"
                     | `Image, _ -> "Image"
-                    | `Video, _ -> "Video" )
+                    | `Video, _ -> "Video"
+                    | `Appx, _ -> "Application/x-directory" )
```
Comment (smondet) [inline src/client/token_viewer.ml:279]: Maybe `"Embedded Web Page"` ?

Shows up there:

screenshot-20220214-143307

File: src/client/import.ml (hunk 1 above)
Comment (marklnichols) [reply src/client/import.ml:449]: I was going to ask about that, I wasn't sure -- that seems reasonable, though.

File: src/client/token_viewer.ml (hunk 2 above)
Comment (marklnichols) [reply src/client/token_viewer.ml:279]: Oh ok, yes need something better there

**Comment by damian0815** [comment]
hi @marklnichols @smondet  - would you consider extending this to cover `text/html` as well? cf #85

File: src/client/import.ml [hunk 3]
```diff
@@ -440,23 +440,27 @@ end
...
       | vid when String.is_prefix vid ~prefix:"video/" ->
           (`Video, String.chop_prefix_exn vid ~prefix:"video/")
+      | app_x when String.equal app_x "application/x-directory" ->
+          (`Appx, String.chop_prefix_exn app_x ~prefix:"application/")
       | other -> Fmt.failwith "Unknown MIME type: %S" other
 
     let to_mime = function
       | `Image, f -> "image/" ^ f
       | `Video, f -> "video/" ^ f
+      | `Appx, f -> "application/x-directory/" ^ f
```
Comment (smondet) [inline src/client/import.ml:463]: then this is going to make `application/x-drectory/x-directory` right?

Comment (marklnichols) [reply src/client/import.ml:463]: true, fixed

**Comment by marklnichols** [comment]
Screenshots for text/html - not the best example as it contains a lot of white space to the sides of the image
image

image

**Comment by smondet** [comment]
https://tzcomet.io/staging/#/token/KT1Fxz4V3LaUcVFpvF8pAAx8G3Z4H7p7hhDg/1
https://tzcomet.io/staging/#/token/KT1WmAHNFxV1JzoGxQYEkE1aFKvNNNSj73sX/10000000003
https://tzcomet.io/staging/#/token/KT1RJ6PbjHpwc3M5rw5s2Nbmefwbuwbdxton/12537

**Pull request merged**